packages/generate-bot/test/generate-bot.js

```diff
@@ -0,0 +1,145 @@
+/**
+ * Copyright 2019 Google LLC. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// define types for a few modules used by probot that do not have their
+// own definitions published. Before taking this step, folks should first
+// check whether type bindings are already published.
+
+const GenerateBot = require("../main.js");
+const { expect } = require("chai");
+const fs = require("fs");
+const recursive = require("recursive-readdir");
+const rimraf = require("rimraf");
+const snapshot = require("snap-shot-it");
+const path = require("path");
+
+const readAllFiles = function(dirNameRead, contentString) {
+  const files = fs.readdirSync(dirNameRead);
+  contentString = contentString || null;
+  files.forEach(function(file) {
+    const readName = path.join(dirNameRead, file);
+    if (fs.statSync(readName).isDirectory()) {
+      readAllFiles(readName, contentString);
+    } else {
+      contentString += fs.readFileSync(readName);
+    }
+  });
+  console.log(contentString);
+  return contentString;
+};
+
+describe("file structure", () => {
+  it("checks that file structure carries over", async () => {
+    const originalStack = await recursive("./templates");
+    GenerateBot.creatingBotFiles("./templates", {
+      programName: "{{programName}}",
+      description: "description",
+      fileLocation: "./tmp"
+    });
+    let createdStack = await recursive("./tmp");
+    createdStack = createdStack.map(contents => {
+      return contents.replace(/tmp/, "templates");
+    });
+    console.log("OG " + originalStack);
+    console.log("CS " + createdStack);
+    expect(originalStack).to.have.members(createdStack);
+  });
+
+  afterEach(() => {
+    rimraf.sync("./tmp");
+  });
+
+  it("checks that the file content carries over", async () => {
+    GenerateBot.creatingBotFiles("./templates", {
+      programName: "helloWorld",
+      description: "says hi",
+      fileLocation: "./helloWorld"
+    });
+
+    const contentString = "Start of snapshot: ";
+    return snapshot(readAllFiles("./helloWorld", contentString));
```

Comment:
try replacing this with:

```
return snapshot(readAllFiles("./helloWorld", contentString).replace('\r\n', '\n'));
```

Windows has `\r\n` on the end of each line rather than `\n`, which might be the source of one of your failing windows tests.